File: src/main/java/kr/mjc/rachel/spring/di/javaconfig/AppleSpeaker.java
package kr.mjc.rachel.spring.di.javaconfig;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class AppleSpeaker implements Speaker {

  public AppleSpeaker() {
    log.debug("appleSpeaker 인스턴스 생성됨");
  }

  @Override
  public void volumeUp() {
    log.debug("appleSpeaker volume up.");
  }

  @Override
  public void volumeDown() {
    log.debug("appleSpeaker volume down.");
  }
}

File: src/main/kotlin/kr/mjc/jacob/spring/springcore/di/javaconfig/AppleSpeaker.kt
package kr.mjc.jacob.spring.springcore.di.javaconfig

import org.slf4j.LoggerFactory

class AppleSpeaker : Speaker {

  private val log = LoggerFactory.getLogger(this.javaClass)

  override fun volumeUp() {
    log.debug("apple speaker volume up")
  }

  override fun volumeDown() {
    log.debug("apple speaker volume down")
  }
}